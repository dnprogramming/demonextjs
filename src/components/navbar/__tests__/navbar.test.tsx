import { fireEvent, render, screen } from '@testing-library/react'
import Navbar from '../navbar';

describe('Page', () => {
    it('renders links', () => {
        render(<Navbar />);

        const homeLink = screen.getByText(/home/i);
        const aboutLink = screen.getByText(/about/i);
        const contactLink = screen.getByText(/contact/i);
        const galleryLink = screen.getByText(/gallery/i);
        const fundraisersLink = screen.getByText(/fundraisers/i);
        const shopLink = screen.getByText(/shop/i);

        expect(homeLink).toBeInTheDocument();
        expect(aboutLink).toBeInTheDocument();
        expect(contactLink).toBeInTheDocument();
        expect(galleryLink).toBeInTheDocument();
        expect(fundraisersLink).toBeInTheDocument();
        expect(shopLink).toBeInTheDocument();
    });

    it('renders hamburger menu', () => {
        render(<Navbar />);
    
        const hamburgerMenu = screen.getByText(/☰/i);
        expect(hamburgerMenu).toBeInTheDocument();
    });

    it('toggles menu on click', () => {
        render(<Navbar />);
    
        const hamburgerMenu = screen.getByText(/☰/i);
        const linksContainer = screen.getByText(/home/i).closest('div');
    
        expect(linksContainer).not.toHaveClass('open');
    
        fireEvent.click(hamburgerMenu);
    
        expect(linksContainer).toHaveClass('open');
    
        fireEvent.click(hamburgerMenu);
    
        expect(linksContainer).not.toHaveClass('open');
    });

  it('renders homepage unchanged', () => {
    const { container } = render(<Navbar />)
    expect(container).toMatchSnapshot()
  })
})