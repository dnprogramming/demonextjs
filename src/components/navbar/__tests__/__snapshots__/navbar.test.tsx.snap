// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Page renders homepage unchanged 1`] = `
<div>
  <div
    class="navbar"
  >
    <div
      class="hamburger"
    >
      â˜°
    </div>
    <div
      class="linksContainer "
    >
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/about"
      >
        About
      </a>
      <a
        href="/contact"
      >
        Contact
      </a>
      <a
        href="/gallery"
      >
        Gallery
      </a>
      <a
        href="/fundraisers"
      >
        Fundraisers
      </a>
      <a
        href="/shop"
      >
        Shop
      </a>
    </div>
  </div>
</div>
`;
